raft/src/main/java/org/apache/kafka/raft/internals/RecordsIterator.java

```diff
@@ -346,29 +343,12 @@ private T decodeDataRecord(Optional<ByteBuffer> key, Optional<ByteBuffer> value)
     }
 
     private static ControlRecord decodeControlRecord(Optional<ByteBuffer> key, Optional<ByteBuffer> value) {
-        if (key.isEmpty()) {
-            throw new IllegalArgumentException("Missing key in the record when a key was expected");
-        } else if (key.get().remaining() == 0) {
+        if (key.get().remaining() == 0) {
```

Comment:
Should we check whether key optional is empty before using get function?

Reply:
agree to @FrankYang0529 - we should keep origin checks